vaadin-crud-flow-parent/vaadin-crud-flow/src/main/java/com/vaadin/flow/component/crud/CrudGrid.java

```diff
@@ -135,20 +136,19 @@ public CrudFilter getFilter() {
      *             CrudFilter&gt;
      */
     @Override
-    public void setDataProvider(DataProvider<E, ?> dataProvider)
-            throws IllegalArgumentException {
+    public GridDataView<E> setItems(DataProvider<E, Void> dataProvider) {
         // Attempt a cast to ensure that the captured ? is actually a CrudFilter
         // Unfortunately this cannot be enforced by the compiler
         try {
-            ConfigurableFilterDataProvider<E, Void, CrudFilter> provider = ((DataProvider<E, CrudFilter>) dataProvider)
+            ConfigurableFilterDataProvider provider = dataProvider
                     .withConfigurableFilter();
```

Comment:
I have some concerns about this: we take a data provider for a Void filter, but then we need it to handle `CrudFilter`.
Does it mean that the developer is required to use unchecked casts on its code?
Or is the `CrudFilter` constraint not required anymore? If this is the case, the comment can be removed

Reply:
Yes, that one is bit problematic because we use `setItems(DataProvider<E, Void> dataProvider)` which is filter-less and then try to apply filter to it by force. 

`CrudFilter` is not a Predicate and cannot be used directly as a filter, but it is used in some examples though.

I didn't have better ideas how to deal with that case.

Reply:
Got it, but I'm still not convinced by this method, since the parameter is a `DataProvider<E, Void> dataProvider` but the javadocs says that it `Sets a DataProvider<E, CrudFilter>`.
But I do not have any hints on how to fix this, since the `DataProvider<E, Void>` comes from the base class.

@tomivirkki what do you think? Is there something that can be done to make the API less confusing for the end user?

Reply:
Grid.setDataProvider kept public for special cases.